core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsBackupResponse.java

```diff
@@ -154,4 +158,28 @@ private Throwable filterException(Throwable throwable) {
       }
       return throwable;
    }
+
+   @Override
+   public void notifyAsyncAck(XSiteAsyncAckListener listener) {
+      XSiteAsyncAckListener nonNullListener = Objects.requireNonNull(listener);
+      for (Map.Entry<XSiteBackup, CompletableFuture<ValidResponse>> entry : backupCalls.entrySet()) {
```

Comment:
Would be worth to wrap the `Map.Entry<XSiteBackup, CompletableFuture<ValidResponse>>` in an object and have some method there to ease understanding and avoid similar code that looks a bit like duplicated?

Reply:
replacing the `Map` with a `List`? yes, it will definitely worth doing it. Probably not in this PR since it is already complex enough xD 
I'll change it in https://issues.jboss.org/browse/ISPN-10192 (currently working on it)

Reply:
nice